Investigate/fix platform_tests/link_flap/test_link_flap.py failures on backend T0

```
Failed: dut port Ethernet124 didn't go down as expected
```
stack trace
```
request = <FixtureRequest for <Function test_link_flap[str2-7050qx-32s-acs-02|Ethernet124]>>
duthosts = [<MultiAsicSonicHost> str2-7050qx-32s-acs-02]
enum_dut_portname = 'str2-7050qx-32s-acs-02|Ethernet124'
fanouthosts = {'str2-7260cx3-acs-fan-20': { os: 'eos', hostname: 'str2-7260cx3-acs-fan-20', device_type: 'FanoutLeaf' }}
@pytest.mark.platform('physical')
def test_link_flap(request, duthosts, enum_dut_portname, fanouthosts):
"""
Validates that link flap works as expected
"""
tlf = TestLinkFlap(request)
dutname, portname = decode_dut_port_name(enum_dut_portname)
for dut in duthosts:
if dutname == 'unknown' or dutname == dut.hostname:
>               tlf.run_link_flap_test(dut, fanouthosts, portname)
dut        = <MultiAsicSonicHost> str2-7050qx-32s-acs-02
duthosts   = [<MultiAsicSonicHost> str2-7050qx-32s-acs-02]
dutname    = 'str2-7050qx-32s-acs-02'
enum_dut_portname = 'str2-7050qx-32s-acs-02|Ethernet124'
fanouthosts = {'str2-7260cx3-acs-fan-20': { os: 'eos', hostname: 'str2-7260cx3-acs-fan-20', device_type: 'FanoutLeaf' }}
portname   = 'Ethernet124'
request    = <FixtureRequest for <Function test_link_flap[str2-7050qx-32s-acs-02|Ethernet124]>>
tlf        = <tests.platform_tests.link_flap.test_link_flap.TestLinkFlap object at 0x7f9750e57510>
platform_tests/link_flap/test_link_flap.py:61:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
platform_tests/link_flap/test_link_flap.py:48: in run_link_flap_test
toggle_one_link(dut, dut_port, fanout, fanout_port)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
dut = <MultiAsicSonicHost> str2-7050qx-32s-acs-02, dut_port = 'Ethernet124'
fanout = { os: 'eos', hostname: 'str2-7260cx3-acs-fan-20', device_type: 'FanoutLeaf' }
fanout_port = 'Ethernet10/1', watch = False
def toggle_one_link(dut, dut_port, fanout, fanout_port, watch=False):
"""
Toggle one link on the fanout.
Args:
dut: DUT host object
dut_port: Port of DUT
fanout: Fanout host object
fanout_port: Port of fanout
watch: Logging system state
"""
logger.info("Testing link flap on %s", dut_port)
pytest_assert(__check_if_status(dut, dut_port, 'up', verbose=True), "Fail: dut port {}: link operational down".format(dut_port))
logger.info("Shutting down fanout switch %s port %s connecting to %s", fanout.hostname, fanout_port, dut_port)
need_recovery = True
try:
fanout.shutdown(fanout_port)
>           pytest_assert(wait_until(30, 1, 0, __check_if_status, dut, dut_port, 'down', True), "dut port {} didn't go down as expected".format(dut_port))
E           Failed: dut port Ethernet124 didn't go down as expected
dut        = <MultiAsicSonicHost> str2-7050qx-32s-acs-02
dut_port   = 'Ethernet124'
fanout     = { os: 'eos', hostname: 'str2-7260cx3-acs-fan-20', device_type: 'FanoutLeaf' }
fanout_port = 'Ethernet10/1'
need_recovery = True
watch      = False
platform_tests/link_flap/link_flap_utils.py:155: Failed
```

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -476,8 +476,15 @@ def fanouthosts(ansible_adhoc, conn_graph_facts, creds, duthosts):
 
                 # Add port name to fanout port mapping port if dut_port is alias.
                 if dut_port in mg_facts['minigraph_port_alias_to_name_map']:
-                    fanout.add_port_map(encode_dut_port_name(
-                       dut_host, mg_facts['minigraph_port_alias_to_name_map'][dut_port]), fanout_port)
+                    mapped_port = mg_facts['minigraph_port_alias_to_name_map'][dut_port]
+                    # only add the mapped port which isn't in device_conn ports to avoid overwriting port map wrongly,
+                    # it happens when an interface has the same name with another alias, for example:
+                    # Interface     Alias
+                    # --------------------
+                    # Ethernet108   Ethernet32
+                    # Ethernet32    Ethernet13/1
+                    if mapped_port not in value.keys():
+                        fanout.add_port_map(encode_dut_port_name(dut_host, mapped_port), fanout_port)
 
                 if dut_host not in fanout.dut_hostnames:
                     fanout.dut_hostnames.append(dut_host)

